datasafe-storage/datasafe-storage-impl-s3/src/main/java/de/adorsys/datasafe/storage/impl/s3/S3StorageService.java

```diff
@@ -28,17 +31,14 @@ public S3StorageService(AmazonS3 s3, String bucketName) {
     @Override
     public Stream<AbsoluteLocation<ResolvedResource>> list(AbsoluteLocation location) {
         log.debug("List at {}", location);
-        ListObjectsV2Request listObjectsV2Request = new ListObjectsV2Request();
-        listObjectsV2Request.setBucketName(bucketName);
         String prefix = location.location().getPath().replaceFirst("^/", "");
-        int len = prefix.length();
-        listObjectsV2Request.setPrefix(prefix);
-        ListObjectsV2Result listObjectsV2Result = s3.listObjectsV2(listObjectsV2Request);
-        List<S3ObjectSummary> objectSummaries = listObjectsV2Result.getObjectSummaries();
-        return objectSummaries.stream()
+
+        S3Objects s3ObjectSummaries = S3Objects.inBucket(s3, bucketName);
```

Comment:
I think proper function is S3Objects.withPrefix

Reply:
yep, but which prefix here I have yo use?

Reply:
yep